src/status_im/contexts/wallet/common/utils_test.cljs

```diff
@@ -31,3 +55,177 @@
+
+(deftest test-calculate-balance
+  (testing "test calculate-balance function"
+    (let [tokens-in-account [{:balances-per-chain         [{:raw-balance "100000000" :chain-id 1}
+                                                           {:raw-balance "50000000" :chain-id 2}
+                                                           {:raw-balance "123456789" :chain-id 3}]
+                              :decimals                   "8"
+                              :market-values-per-currency {:usd {:price 1}}}
+                             {:balances-per-chain         [{:raw-balance "200000000" :chain-id 1}
+                                                           {:raw-balance "100000000" :chain-id 2}
+                                                           {:raw-balance "987654321" :chain-id 3}]
+                              :decimals                   "8"
+                              :market-values-per-currency {:usd {:price 1}}}]]
+      (is (= (utils/calculate-balance tokens-in-account) 10234567.89)))
+
+    (let [tokens-in-account [{:balances-per-chain         [{:raw-balance "0" :chain-id 1}
+                                                           {:raw-balance "0" :chain-id 2}
+                                                           {:raw-balance "0" :chain-id 3}]
+                              :decimals                   "4"
+                              :market-values-per-currency {:usd {:price 1}}}
+                             {:balances-per-chain         [{:raw-balance "0" :chain-id 1}
+                                                           {:raw-balance "0" :chain-id 2}
+                                                           {:raw-balance "0" :chain-id 3}]
+                              :decimals                   "4"
+                              :market-values-per-currency {:usd {:price 1}}}]]
+      (is (= (utils/calculate-balance tokens-in-account) 0.0)))))
+
+(deftest test-network-list
+  (testing "network-list function"
+    (let [balances-per-chain {:chain-1 {:balance 100}
+                              :chain-2 {:balance 200}
+                              :chain-3 {:balance 300}}
+          networks           [{:chain-id :chain-1 :name "Network A"}
+                              {:chain-id :chain-2 :name "Network B"}
+                              {:chain-id :chain-3 :name "Network C"}]]
+      (is (= (utils/network-list {:balances-per-chain balances-per-chain} networks)
+             #{:chain-1 :chain-2 :chain-3})))
+
+    (let [balances-per-chain {:chain-1 {:balance 100}
+                              :chain-2 {:balance 200}
+                              :chain-3 {:balance 300}}
+          networks           [{:chain-id :chain-1 :name "Network A"}
+                              {:chain-id :chain-3 :name "Network C"}]]
+      (is (= (utils/network-list {:balances-per-chain balances-per-chain} networks)
+             #{:chain-1 :chain-3})))))
```

Comment:
just curious is `:chain-1` etc abitrary? this is normally the chain id? e.g `:1` ?

Reply:
@J-Son89 yes the chain id like `:1`. I think the value used in the test doesn't matter much as long as the test returns the correct value, but I can't update it to `:1` 👍

Reply:
actually we are transforming (or we should) to `1` instead.

it is important to keep the type because a function could break if we don't provide the proper one (such as the common exception `name not supported`).